DanceCat/FrequencyTaskChecker.py

```
@@ -104,14 +108,24 @@ def start(interval=60, pid_path='frequency.pid'):
     :param interval: Seconds the checker will sleep throughout idle time.
     :param pid_path: Location of the PID file.
     """
+
     try:
-        pid_file = open(pid_path, 'r')
-        pid = int(pid_file.read())
-        print(
-            "Frequency Task Checker is already started with pid {pid}".
-            format(pid=pid)
-        )
-        pid_file.close()
+        with open(pid_path, 'r') as fp:
+            pid = int(fp.read())
+            if pid and psutil.pid_exists(pid):
+                print(
+                    'Frequency Task Checker is already started with pid {pid}'.
+                    format(pid=pid)
+                )
+            elif pid and not psutil.pid_exists(pid):
+                print('Removing dead pid file')
+                os.remove(pid_path)
```

Should we raise an Exception here? Let's discuss more:
- If we want to start the process if not exists, raise an Exception here for later code to catch it and start the process.
- If we want run this process as separate service, a signal controller system maybe need.